src/backend/api/Fusion.Resources.Logic/Requests/Commands/ResourceAllocationRequest/JointVenture/Create.cs

```diff
@@ -0,0 +1,335 @@
+                {
+                    AdditionalNote = note;
+                    return this;
+                }
+
+                public Create WithProposedChanges(Dictionary<string, object>? changes)
+                {
+                    ProposedChanges = changes;
+                    return this;
+                }
+
+                public Create WithPositionInstance(Guid basePositionId, DateTime from, DateTime to, double? workload,
+                    string? obs, Guid? locationId)
+                {
+                    OrgPositionInstance = new Domain.ResourceAllocationRequest.QueryPositionInstance
+                    {
+                        Id = basePositionId,
+                        Workload = workload,
+                        AppliesFrom = @from,
+                        AppliesTo = to,
+                        Obs = obs,
+                        LocationId = locationId
+                    };
+
+                    return this;
+                }
+
+                public class Validator : AbstractValidator<Create>
+                {
+                    public Validator()
+                    {
+                        RuleFor(x => x.AssignedDepartment).NotContainScriptTag().MaximumLength(500);
+                        RuleFor(x => x.Discipline).NotContainScriptTag().MaximumLength(500);
+                        RuleFor(x => x.AdditionalNote).NotContainScriptTag().MaximumLength(5000);
+
+                        RuleFor(x => x.OrgPositionId).NotEmpty().When(x => x.OrgPositionId != null);
+                        RuleFor(x => x.OrgPositionInstance).NotNull();
+                        RuleFor(x => x.OrgPositionInstance).SetValidator(PositionInstanceValidator)
+                            .When(x => x.OrgPositionInstance != null);
+                        RuleFor(x => x.ProposedChanges).SetValidator(ProposedChangesValidator)
+                            .When(x => x.ProposedChanges != null);
+
+                        RuleFor(x => x.ProposedPersonAzureUniqueId).NotEmpty()
+                            .When(x => x.ProposedPersonAzureUniqueId != null);
+
+                        RuleFor(x => x.OrgProjectId).NotNull();
+                        RuleFor(x => x.IsDraft).NotNull();
+                    }
+
+                    private static IPropertyValidator ProposedChangesValidator =>
+                        new CustomValidator<Dictionary<string, object>>(
+                            (prop, context) =>
+                            {
+                                foreach (var k in prop.Keys.Where(k => k.Length > 100))
+                                {
+                                    context.AddFailure(new ValidationFailure($"{context.PropertyName}.key",
+                                        "Key cannot exceed 100 characters", k));
+                                }
+
+                            });
+
+                    private static IPropertyValidator PositionInstanceValidator =>
```

Comment:
Should maybe look in to defining typed validators to this can be reused?

Could move this stuff out to extension methods, like the NotContainScript one?